modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/snapshot/IgniteSnapshotManager.java

```diff
@@ -394,6 +403,13 @@ public class IgniteSnapshotManager extends GridCacheSharedManagerAdapter
     /** Snapshot transfer rate limit in bytes/sec. */
     private final DistributedLongProperty snapshotTransferRate = detachedLongProperty(SNAPSHOT_TRANSFER_RATE_DMS_KEY);
 
+    /**
+     * Snapshot delta sort batch size in pages count. If set then delta pages will be indexed by page index to almost
+     * sequential disk write on apply. This generates an extra delta read. If value is non-positive or not set then delta
+     * pages will be applied directly.
+     */
+    private final DistributedIntegerProperty deltaSortBatch = detachedIntegerProperty(SNAPSHOT_DELTA_SORT_BATCH_SIZE_KEY);
```

Comment:
This will not work if encryption is enabled, right?

Reply:
Encryption works by pages, so ordering will work.